### PR TITLE
Add buildifier linter diagnostics

### PR DESCRIPTION
**Pull request by kgreenek**
Closes https://github.com/jose-elias-alvarez/null-ls.nvim/issues/734

### CODE REVIEW

File: lua/null-ls/builtins/diagnostics/buildifier.lua [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    return diagnostics
+end
+
+local function parse_error(line)
+    local pattern = [[.-:(%d+):(%d+): (.*)]]
+    local results = { line:match(pattern) }
+    local row = tonumber(results[1])
+    local col = tonumber(results[2])
+    -- Remove trailing newline in the error message.
+    local message = rtrim(results[3])
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/buildifier.lua:53]: Would `vim.trim` achieve the same result here?

**Review by jose-elias-alvarez — COMMENTED** [review]
Maybe one small optimization we can make but other than that this looks good.

File: lua/null-ls/builtins/diagnostics/buildifier.lua (hunk 1 above)
Comment (kgreenek) [reply lua/null-ls/builtins/diagnostics/buildifier.lua]: Nice! Looks like it does. I just didn't know about that function :)

**Comment by jose-elias-alvarez** [comment]
Great, thanks!

**Pull request merged**